packages/ember-glimmer/lib/syntax.js

```diff
@@ -0,0 +1,36 @@
+import { RenderSyntax } from './syntax/render';
+import { OutletSyntax } from './syntax/outlet';
+import { MountSyntax } from './syntax/mount';
+import { DynamicComponentSyntax } from './syntax/dynamic-component';
+
+let syntaxKeys = [];
+let syntaxes = [];
+
+export function registerSyntax(key, syntax) {
+  syntaxKeys.push(key);
+  syntaxes.push(syntax);
+}
+
+export function findSyntaxBuilder(key) {
+  let index = syntaxKeys.indexOf(key);
+
+  if (index > -1) {
+    return syntaxes[index];
+  }
+}
+
+registerSyntax('render', function(options) {
+  return new RenderSyntax(options);
+});
+
+registerSyntax('outlet', function(options) {
+  return new OutletSyntax(options);
+});
+
+registerSyntax('mount', function(options) {
+  return new MountSyntax(options);
+});
+
+registerSyntax('component', function(options) {
+  return DynamicComponentSyntax.create(options);
```

Comment:
We can probably just make all syntax have a `create` that takes care of arguments normalization, assertions etc, like in https://github.com/emberjs/ember.js/pull/14172.

Then we can just do `registerSyntax('component', DynamicComponentSyntax)`, etc, and call `.create` on the other side.